pkg/controller/statusmanager/status_manager.go

```diff
@@ -203,12 +214,86 @@ func (status *StatusManager) set(reachedAvailableLevel bool, conditions ...operv
 		log.Printf("Failed to set operator status: %v", err)
 	}
 
+	if status.hypershiftConfig.Enabled {
+		err = retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+			hcp := &hyperv1.HostedControlPlane{ObjectMeta: metav1.ObjectMeta{Name: status.hypershiftConfig.Name}}
+			gvr := schema.GroupVersionResource{
+				Group:    "hypershift.openshift.io",
+				Version:  "v1alpha1",
+				Resource: "hostedcontrolplanes",
+			}
+			hcpObj, err := status.client.ClientFor(cnoclient.ManagementClusterName).Dynamic().Resource(gvr).Namespace(
```

Comment:
is the dynamic client needed here? could we just use `CRClient` instead?

Reply:
let me try CRClient, I remembered it didn't work with the route PR.

Reply:
`CRClient` works.
Removed dynamic client.